src/Quartz/Simpl/RAMJobStore.cs

```diff
@@ -1611,6 +1611,8 @@ protected virtual bool ApplyMisfire(TriggerWrapper tw)
     {
         lock (lockObject)
         {
+            using DateTimeScope _ = new();
```

Comment:
isn't this the same thing as just having same `now` value during having the lock?

Reply:
had a quick look and doesn't seem to be enough. I'm not fond of adding complexity to our SystemTime hekoer as it will be later replaced with TimeProvider. Should probably investigate further inside RAMJobstore and also how this reflects to AdoJobStore.

Reply:
> had a quick look and doesn't seem to be enough. I'm not fond of adding complexity to our SystemTime hekoer as it will be later replaced with TimeProvider. Should probably investigate further inside RAMJobstore and also how this reflects to AdoJobStore.

Do you have a specific vision for updating the code then?